### PR TITLE
Add 0.193 release notes

### PR DESCRIPTION
**Pull request by nezihyigitbasi**

### CODE REVIEW

File: presto-docs/src/main/sphinx/release/release-0.193.rst [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+General Changes
+---------------
+
+* Add :func:`map_zip_with` lambda function.
+* Add :func:`normal_cdf` function.
+* Add ``SET_DIGEST`` type and related functions.
+* Fix ``Tried to free more revocable memory than is reserved`` error for queries
+  that enable spilling and that are moved to the reserved memory pool.
+* Fix a bug that can cause the planner to go into an infinite loop. This bug was affecting queries
```
Comment (martint) [inline presto-docs/src/main/sphinx/release/release-0.193.rst]: Move the bug fixes to the top. I generally put bug fixes, performance improvements and new features in that order.

File: presto-docs/src/main/sphinx/release/release-0.193.rst [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+Release 0.193
+=============
+
+General Changes
+---------------
+
+* Add :func:`map_zip_with` lambda function.
+* Add :func:`normal_cdf` function.
+* Add ``SET_DIGEST`` type and related functions.
+* Fix ``Tried to free more revocable memory than is reserved`` error for queries
```
Comment (martint) [inline presto-docs/src/main/sphinx/release/release-0.193.rst]: Fix ... error for queries that run in the reserved memory pool with spilling enabled.

File: presto-docs/src/main/sphinx/release/release-0.193.rst (hunk 1 above)
Comment (martint) [reply presto-docs/src/main/sphinx/release/release-0.193.rst]: Fix an infinite loop during planning for queries containing non-trivial predicates.

File: presto-docs/src/main/sphinx/release/release-0.193.rst [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+  that enable spilling and that are moved to the reserved memory pool.
+* Fix a bug that can cause the planner to go into an infinite loop. This bug was affecting queries
+  with multiple predicates that couldn't be pushed down to the connectors.
+* Fix a bug that may cause queries with ``row_number()`` to fail or return incorrect results.
```
Comment (martint) [inline presto-docs/src/main/sphinx/release/release-0.193.rst]: Fix a bug that causes queries that constrain the result of ``row_number()`` to numbers less than or equal to 0 to fail or return incorrect results.

File: presto-docs/src/main/sphinx/release/release-0.193.rst [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+  If the query had an equality predicate against ``-1`` or ``0`` the bug would cause the query to fail.
+  If the query had a less-than-or-equal-to predicate against a negative value the bug would cause
+  the query to produce incorrect results.
+* Fix query analysis failure when lambda is used in ``UNNEST`` or ``VALUES``.
```
Comment (martint) [inline presto-docs/src/main/sphinx/release/release-0.193.rst]: Fix failure during query planning when lambda expressions are used in ``UNNEST`` or ``VALUES`` clauses.

File: presto-docs/src/main/sphinx/release/release-0.193.rst [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+* Add the peak of total non-revocable memory reservation (user + system memory) to query stats.
```
Comment (martint) [inline presto-docs/src/main/sphinx/release/release-0.193.rst]: "non-revocable" doesn't mean anything to users. I'd just rephrase as "Add query stat to keep track of peak total memory".

File: presto-docs/src/main/sphinx/release/release-0.193.rst [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+* Add an optimizer rule that rewrites table scans in queries against partitioned tables to an empty relation
```
Comment (martint) [inline presto-docs/src/main/sphinx/release/release-0.193.rst]: Improve performance of queries that filter all data from a table up-front (e.g., due to partition pruning).

File: presto-docs/src/main/sphinx/release/release-0.193.rst [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+  when the query has predicate(s) that do not match any partitions of the table.
+* Improve the performance of the :func:`ST_Contains` function.
+* Turn on new local scheduling algorithm by default (described in 0.181 release notes).
```
Comment (martint) [inline presto-docs/src/main/sphinx/release/release-0.193.rst]: Replace "described in 0.181 release notes" with "(See <link to 0.181 notes>)"

File: presto-docs/src/main/sphinx/release/release-0.193.rst [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+* Remove the ``information_schema.__internal_partitions__`` table. This table was only useful
+  for the Hive connector and it became obsolete after ``SHOW PARTITIONS`` has been improved.
+
+Security Changes
+----------------
+
+* When using multiple authentication types the authentication methods are now applied in the order
```
Comment (martint) [inline presto-docs/src/main/sphinx/release/release-0.193.rst]: Apply the authentication methods in the order they are listed in the http-server.authentication.type configuration.

File: presto-docs/src/main/sphinx/release/release-0.193.rst [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+  If the query had an equality predicate against ``-1`` or ``0`` the bug would cause the query to fail.
+  If the query had a less-than-or-equal-to predicate against a negative value the bug would cause
+  the query to produce incorrect results.
+* Fix query analysis failure when lambda is used in ``UNNEST`` or ``VALUES``.
+* Add the peak of total non-revocable memory reservation (user + system memory) to query stats.
+* Add an optimizer rule that rewrites table scans in queries against partitioned tables to an empty relation
+  when the query has predicate(s) that do not match any partitions of the table.
+* Improve the performance of the :func:`ST_Contains` function.
+* Turn on new local scheduling algorithm by default (described in 0.181 release notes).
+* Remove the ``information_schema.__internal_partitions__`` table. This table was only useful
```
Comment (martint) [inline presto-docs/src/main/sphinx/release/release-0.193.rst]: I'd remove the comment about SHOW PARTITIONS

File: presto-docs/src/main/sphinx/release/release-0.193.rst [hunk 10]
```diff
@@ -0,0 +1,76 @@
...
+Security Changes
+----------------
+
+* When using multiple authentication types the authentication methods are now applied in the order
+  they are specified in the ``http-server.authentication.type`` configuration.
+
+CLI Changes
+-----------
+
+* Fix CLI to abort the query when the pager exits.
```
Comment (martint) [inline presto-docs/src/main/sphinx/release/release-0.193.rst]: Is this about sending a cancellation to the server when the pager exits? Maybe rephrase as "Abort the query correctly when the result pager exits"

File: presto-docs/src/main/sphinx/release/release-0.193.rst [hunk 11]
```diff
@@ -0,0 +1,76 @@
...
+* Fix rendering of maps of Bing tiles in the CLI.
```
Comment (martint) [inline presto-docs/src/main/sphinx/release/release-0.193.rst]: remove "in the CLI"

File: presto-docs/src/main/sphinx/release/release-0.193.rst [hunk 12]
```diff
@@ -0,0 +1,76 @@
...
+------------------------
+
+* Fix a bug in ``ALTER TABLE ... RENAME TO ...`` that was causing the source schema to be used if the
+  destination schema was different the source schema.
+* Fix a bug that allowed creating tables in a nonexistent schema.
+
+MongoDB Changes
+---------------
+
+* Fix ``INSERT INTO`` support.
```
Comment (martint) [inline presto-docs/src/main/sphinx/release/release-0.193.rst]: What was broken?

File: presto-docs/src/main/sphinx/release/release-0.193.rst [hunk 13]
```diff
@@ -0,0 +1,76 @@
...
+-------------------
+
+* Use SSL by default for port 443.
+
+Hive Changes
+------------
+
+* Fix a bug that causes a ``ConcurrentModificationException`` when dropping any column
+  other than the last one.
+* Fix a bug that causes a newly added partition to have incorrect on-disk data format
```
Comment (martint) [inline presto-docs/src/main/sphinx/release/release-0.193.rst]: What's "non-default text format"?

File: presto-docs/src/main/sphinx/release/release-0.193.rst (hunk 12 above)
Comment (nezihyigitbasi) [reply presto-docs/src/main/sphinx/release/release-0.193.rst]: In https://github.com/prestodb/presto/issues/9802 @kokosing says

> INSERT INTO did not work for mongodb at all, as there was no insert related method in connector, see #9736.

File: presto-docs/src/main/sphinx/release/release-0.193.rst (hunk 13 above)
Comment (nezihyigitbasi) [reply presto-docs/src/main/sphinx/release/release-0.193.rst]: /cc @findepi

Comment (nezihyigitbasi) [reply presto-docs/src/main/sphinx/release/release-0.193.rst]: IIUC this bug affects only the text file format. So I think we can just say `Fix a bug that causes a newly added partition to have incorrect on-disk data format for tables using the text file format`.

The related issue and PR are:
- https://github.com/prestodb/presto/issues/7301
- https://github.com/prestodb/presto/pull/9784

Comment (findepi) [reply presto-docs/src/main/sphinx/release/release-0.193.rst]: Yup. It was about text format, in case when non-default format properties (e.g. field delimiter) were supposed to be used.

File: presto-docs/src/main/sphinx/release/release-0.193.rst (hunk 12 above)
Comment (findepi) [reply presto-docs/src/main/sphinx/release/release-0.193.rst]: Exactly. INSERT didn't work at all, since related API methods were implemented only partially (finish was not implemented).

File: presto-docs/src/main/sphinx/release/release-0.193.rst (hunk 5 above)
Comment (findepi) [reply presto-docs/src/main/sphinx/release/release-0.193.rst]: or you can leave this bullet out altogether?

Comment (nezihyigitbasi) [reply presto-docs/src/main/sphinx/release/release-0.193.rst]: I added this because I see some past release notes have similar items.

File: presto-docs/src/main/sphinx/release/release-0.193.rst (hunk 12 above)
Comment (martint) [reply presto-docs/src/main/sphinx/release/release-0.193.rst]: So then, "Add support for INSERT INTO"?

**Comment by findepi** [comment]
indeed

File: presto-docs/src/main/sphinx/release/release-0.193.rst (hunk 12 above)
Comment (kokosing) [reply presto-docs/src/main/sphinx/release/release-0.193.rst]: It looks like at some point in the time it was supported, but there were no tests... and you know what happened next.

Comment (nezihyigitbasi) [reply presto-docs/src/main/sphinx/release/release-0.193.rst]: Looking at the commits I guess `INSERT INTO` stopped working after the legacy connector API is removed in `0.155`, which was calling `commitInsert()` that the connector was already implementing. When the legacy API is removed `finishInsert()` is called by the engine, so it's probably broken since `0.155`.

I also find the current version of this release note item vague, an alternative is to give some context:

`* Fix ``INSERT INTO`` support, which was broken since the removal of the legacy connector API in :doc:`release-0.155`.`

File: presto-docs/src/main/sphinx/release/release-0.193.rst [hunk 14]
```diff
@@ -0,0 +1,71 @@
+=============
+Release 0.193
+=============
+
+General Changes
+---------------
+
+* Fix an infinite loop during planning for queries containing non-trivial predicates.
+* Fix a bug that causes queries that constrain the result of ``row_number()`` to numbers less than or
+  equal to 0 to fail or return incorrect results.
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-0.193.rst]: Change this to "less than one"

File: presto-docs/src/main/sphinx/release/release-0.193.rst [hunk 15]
```diff
@@ -0,0 +1,71 @@
...
+----------------
+
+* Apply the authentication methods in the order they are listed in the
+  ``http-server.authentication.type`` configuration.
+
+CLI Changes
+-----------
+
+* Fix rendering of maps of Bing tiles.
+* Abort the query correctly when the result pager exits.
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-0.193.rst]: What does "correctly" mean? It's either aborted or not. Remove this word.

File: presto-docs/src/main/sphinx/release/release-0.193.rst (hunk 13 above)
Comment (electrum) [reply presto-docs/src/main/sphinx/release/release-0.193.rst]: This needs to be more specific:

* Correctly write files for text format tables that use non-default delimiters. Previously, they were written with the default delimiter.

File: presto-docs/src/main/sphinx/release/release-0.193.rst [hunk 16]
```diff
@@ -0,0 +1,71 @@
...
+* Use SSL by default for port 443.
+
+Hive Changes
+------------
+
+* Fix a bug that causes a ``ConcurrentModificationException`` when dropping any column
+  other than the last one.
+* Fix a bug that causes a newly added partition to have incorrect on-disk data format
+  for tables using the text file format.
+* Fix a bug in ``PrestoS3FileSystem`` that causes queries to fail if they read data from
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-0.193.rst]: `PrestoS3FileSystem` is an internal implementation detail. All users know is about reading data from S3.

* Fix reading data from S3 if the data is in a region other than ...

File: presto-docs/src/main/sphinx/release/release-0.193.rst [hunk 17]
```diff
@@ -0,0 +1,71 @@
...
+  regions other than ``us-east-1``. Previously, such queries would fail with
+  ``"The authorization header is malformed; the region 'us-east-1' is wrong; expecting '<region_name>'"``,
+  where ``<region_name>`` is the S3 region hosting the bucket that is queried.
+* Enable ``SHOW PARTITIONS FROM <table> WHERE <condition>`` to work for tables
+  that have more than ``hive.max-partitions-per-scan`` partitions as long as
+  the specified ``<condition>`` reduces the number of partitions below this limit.
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-0.193.rst]: below -> "to be below"

File: presto-docs/src/main/sphinx/release/release-0.193.rst [hunk 18]
```diff
@@ -0,0 +1,71 @@
...
+
+Blackhole Changes
+-----------------
+
+* Fix a bug that allows creating tables in a nonexistent schema.
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-0.193.rst]: * Do not allow creating tables in a nonexistent schema.

**Comment by electrum** [comment]
These release notes have way too many "fix a bug". There's usually a better way to write it.

Bad: Fix a bug that causes queries to fail when ...
Good: Prevent queries from failing when ...

File: presto-docs/src/main/sphinx/release/release-0.193.rst [hunk 19]
```diff
@@ -0,0 +1,71 @@
...
+* Add support for ``CREATE SCHEMA``.
+
+Memory Connector Changes
+------------------------
+
+* Fix a bug in ``ALTER TABLE ... RENAME TO ...`` that causes the source schema to be used if the
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-0.193.rst]: * Allow renaming tables across schemas. Previously, the target schema was ignored.

File: presto-docs/src/main/sphinx/release/release-0.193.rst [hunk 20]
```diff
@@ -0,0 +1,71 @@
...
+  destination schema was different than the source schema.
+* Fix a bug that allows creating tables in a nonexistent schema.
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-0.193.rst]: * Do not allow creating tables in a nonexistent schema.

File: presto-docs/src/main/sphinx/release/release-0.193.rst (hunk 12 above)
Comment (electrum) [reply presto-docs/src/main/sphinx/release/release-0.193.rst]: The command is ``INSERT``, not ``INSERT INTO`` (see it's title in the manual).

```
* Add ``INSERT`` support. It was previously removed in 0.155.

File: presto-docs/src/main/sphinx/release/release-0.193.rst [hunk 21]
```diff
@@ -0,0 +1,71 @@
...
+
+JDBC Driver Changes
+-------------------
+
+* Use SSL by default for port 443.
+
+Hive Changes
+------------
+
+* Fix a bug that causes a ``ConcurrentModificationException`` when dropping any column
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-0.193.rst]:
```
* Allow dropping any column in a table. Previously, dropping columns other
  than the last one would fail with ``ConcurrentModificationException``.

File: presto-docs/src/main/sphinx/release/release-0.193.rst [hunk 22]
```diff
@@ -0,0 +1,71 @@
...
+Security Changes
+----------------
+
+* Apply the authentication methods in the order they are listed in the
+  ``http-server.authentication.type`` configuration.
+
+CLI Changes
+-----------
+
+* Fix rendering of maps of Bing tiles.
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-0.193.rst]: What exactly is fixed? What was the previous behavior?

File: presto-docs/src/main/sphinx/release/release-0.193.rst [hunk 23]
```diff
@@ -0,0 +1,71 @@
...
+* Fix failure during query planning when lambda expressions are used in ``UNNEST`` or ``VALUES`` clauses.
+* Fix ``Tried to free more revocable memory than is reserved`` error for queries that run in the
+  reserved memory pool with spilling enabled.
+* Improve the performance of the :func:`ST_Contains` function.
+* Add :func:`map_zip_with` lambda function.
+* Add :func:`normal_cdf` function.
+* Add ``SET_DIGEST`` type and related functions.
+* Add query stat to keep track of peak total memory.
+* Improve performance of queries that filter all data from a table up-front (e.g., due to partition pruning).
+* Turn on new local scheduling algorithm by default (see :doc:`release-0.181`)
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-0.193.rst]: Missing period at end

File: presto-docs/src/main/sphinx/release/release-0.193.rst [hunk 24]
```diff
@@ -0,0 +1,71 @@
...
+* Fix a bug that causes queries that constrain the result of ``row_number()`` to numbers less than or
+  equal to 0 to fail or return incorrect results.
+* Fix failure during query planning when lambda expressions are used in ``UNNEST`` or ``VALUES`` clauses.
+* Fix ``Tried to free more revocable memory than is reserved`` error for queries that run in the
+  reserved memory pool with spilling enabled.
+* Improve the performance of the :func:`ST_Contains` function.
+* Add :func:`map_zip_with` lambda function.
+* Add :func:`normal_cdf` function.
+* Add ``SET_DIGEST`` type and related functions.
+* Add query stat to keep track of peak total memory.
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-0.193.rst]: This phrasing is a bit strange. How about

* Add query stat that tracks peak total memory.

**Comment by nezihyigitbasi** [comment]
I addressed @electrum's comment and updated the mongodb connector release notes as I commented above.

File: presto-docs/src/main/sphinx/release/release-0.193.rst [hunk 25]
```diff
@@ -0,0 +1,71 @@
...
+* Fix an infinite loop during planning for queries containing non-trivial predicates.
+* Fix a bug that causes queries that constrain the result of ``row_number()`` to numbers less than or
+  equal to 0 to fail or return incorrect results.
+* Fix failure during query planning when lambda expressions are used in ``UNNEST`` or ``VALUES`` clauses.
+* Fix ``Tried to free more revocable memory than is reserved`` error for queries that run in the
+  reserved memory pool with spilling enabled.
+* Improve the performance of the :func:`ST_Contains` function.
+* Add :func:`map_zip_with` lambda function.
+* Add :func:`normal_cdf` function.
+* Add ``SET_DIGEST`` type and related functions.
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-0.193.rst]: Are these documented?

File: presto-docs/src/main/sphinx/release/release-0.193.rst [hunk 26]
```diff
@@ -0,0 +1,71 @@
...
+=============
+
+General Changes
+---------------
+
+* Fix an infinite loop during planning for queries containing non-trivial predicates.
+* Fix a bug that causes queries that constrain the result of ``row_number()`` to numbers less than or
+  equal to 0 to fail or return incorrect results.
+* Fix failure during query planning when lambda expressions are used in ``UNNEST`` or ``VALUES`` clauses.
+* Fix ``Tried to free more revocable memory than is reserved`` error for queries that run in the
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-0.193.rst]: "... for queries that have spilling enabled and run in the reserved memory pool"

**Comment by electrum** [comment]
I just added a bunch of comments. Please expand all of the "show outdated" links

File: presto-docs/src/main/sphinx/release/release-0.193.rst (hunk 22 above)
Comment (nezihyigitbasi) [reply presto-docs/src/main/sphinx/release/release-0.193.rst]: maps of bing tiles couldn't be rendered on the presto-cli: https://github.com/prestodb/presto/issues/9693

File: presto-docs/src/main/sphinx/release/release-0.193.rst [hunk 27]
```diff
@@ -0,0 +1,71 @@
...
+------------------------
+
+* Fix ``ALTER TABLE ... RENAME TO ...`` to use the destination schema if the destination schema
+  is different than the source schema.
+* Prevent creating tables in a nonexistent schema.
+
+MongoDB Changes
+---------------
+
+* Fix ``INSERT INTO`` support, which was broken since the removal of the legacy connector API in :doc:`release-0.155`.
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-0.193.rst]: Only link to previous release notes if there's a direct reason someone would want to read it (such as for the scheduling algorithm change as above, since it is discussed in those release notes). In this case, there's nothing about this removal / breakage in those release notes.

The command is named `INSERT`, not `INSERT INTO` (see title of the command in the manual).

I would avoid saying "fix" and "broken" here since those are ambiguous. They could mean it inserts partial results, fails for certain data types, etc.

```
* Add ``INSERT`` support. It was previously removed in 0.155.

File: presto-docs/src/main/sphinx/release/release-0.193.rst [hunk 28]
```diff
@@ -0,0 +1,70 @@
+=============
+Release 0.193
+=============
+
+General Changes
+---------------
+
+* Fix an infinite loop during planning for queries containing non-trivial predicates.
+* Fix a bug that causes queries that constrain the result of ``row_number()`` to numbers less than one
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-0.193.rst]: How about
```
* Fix ``row_number()`` optimization that causes query failure or incorrect results
  for queries that constrain the result of ``row_number()`` to be less than one.
